src/hyperlight_host/src/hypervisor/kvm.rs

```diff
@@ -101,15 +638,36 @@ impl KVMDriver {
         let mut vcpu_fd = vm_fd.create_vcpu(0)?;
         Self::setup_initial_sregs(&mut vcpu_fd, pml4_addr)?;
 
+        #[cfg(gdb)]
+        let (debug, gdb_conn) = if let Some(DebugInfo { port }) = debug_info {
+            (
+                Some(debug::KvmDebug::new()),
+                Some(create_gdb_thread(*port).map_err(|_| new_error!("Cannot create GDB thread"))?),
```

Comment:
To me it doesn't seem right to create the gdb thread from the KVM driver. In the future, we might support debugging on mshv, in which case we'd probably want to create this after the particular driver has been created. Do you think it's possible to move this somewhere else?

Reply:
It doesn't need to be there anymore, since the big change.
However, the specific hypervisor would have to own the connection to `gdb` for it to be able to receive/send messages.